android/KMEA/app/src/main/assets/android-host.js

```diff
@@ -199,17 +193,14 @@ function setNumericLayer() {
 }
 
 function updateKMText(text) {
-  var ta = document.getElementById('ta');
-  console_debug('updateKMText(text='+text+') ta.value='+ta.value);
-
   if(text == undefined) {
       text = '';
   }
 
-  if(ta.value != text) {
-    ta.value = text;
-    window.resetContext();
-  }
+  console_debug('updateKMText(text='+text+') context.value='+keyman.context.getText());
+
+  keyman.context.setText(text);
+  keyman.resetContext();
```

Comment:
I don't think we should be resetting context here, should we, unless text has changed? What happens to deadkeys?